test/unit/specs/components/Identity/index.spec.js

```diff
@@ -59,5 +61,61 @@ describe("Identity", () => {
         expect(getEcid()).toBeUndefined();
       });
     });
+
+    describe("onResponse", () => {
+      let cookieJar;
+      let response;
+      beforeEach(() => {
+        cookieJar = {
+          get(key) {
+            return key === "ECID" ? "ABC" : null;
+          },
+          set: jasmine.createSpy()
+        };
+        response = {
+          getPayloadsByType: jasmine
+            .createSpy()
+            .and.returnValue([{ id: "ABC" }]),
+          toJSON: jasmine.createSpy()
```

Comment:
Don't feel like you need to stub all methods on your mock objects. If `toJSON` isn't used in the code under test, I feel like it's fine to not fill it in here.